File: dailyChallenges/0015-IsomorphicString/15-IsomorphicString.go
// In golang, strings are immutable
// we are not allowed to change the individual characters of a string directly.
// Using bytes allows us to work with mutable data : )
func isIsomorphic(s string, t string) bool {
	mapST := make(map[byte]byte)
	mapTS := make(map[byte]byte)

	if len(s) != len(t) {
		return false
	}

	for i := 0; i < len(s); i++ {
		c1 := s[i]
		c2 := t[i]

		if (mapST[c1] != 0 && mapST[c1] != c2) || (mapTS[c2] != 0 && mapTS[c2] != c1) {
			return false
		} else {
			mapST[c1] = c2
			mapTS[c2] = c1
		}
	}
	return true
}

File: dailyChallenges/0015-IsomorphicString/15-IsomorphicString.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map <char, char> mapST;
        unordered_map <char, char> mapTS;
        char c1, c2;

        if(s.length() != t.length())
        {
            return false;
        }

        for(int i=0; i<s.length(); i++)
        {
            c1 = s[i];
            c2 = t[i];

            if ((mapST.find(c1) != mapST.end() && mapST[c1] != c2) || (mapTS.find(c2) != mapTS.end() && mapTS[c2] != c1)) 
            {
                return false;
            }
            else
            {
                mapST[c1] = c2;
                mapTS[c2] = c1;
            }
        }
        return true;
    }
};